### DISCUSSION TITLE
GitHub Actions API support for setting default Workflow permissions

### DISCUSSION THREAD

**Discussion by HariSekhon** [community/community#14418] [Enterprise] [answered]
Please can an API endpoint be added for setting default Workflows permissions - both at the org and individual repo levels.

Currently this is a big limitation in automating security lockdown of a GitHub Enterprise account and also is limiting the official GitHub Terraform provider (integrations/github) provider which can't support this until the API does.

**Comment by ethomson** [comment]
Thanks for the feedback @HariSekhon - I'll make sure that this is captured on the backlog.

**Comment by felix11h** [comment, marked as answer]
If I read this question correctly, an endpoint for organizations is now available here, https://docs.github.com/en/rest/actions/permissions#set-default-workflow-permissions-for-an-organization.

**Reply by HariSekhon** [reply]
Excellent, thanks for posting this!

**Comment by HariSekhon** [comment]
I've been waiting on this to request that the Terraform GitHub provider supports this, which I've just raised now that the API supports this:

https://github.com/integrations/terraform-provider-github/issues/1265

https://github.com/integrations/terraform-provider-github/issues/1266

related to better management of GitHub organizations in Terraform here:

https://github.com/integrations/terraform-provider-github/issues/1200
